### PR TITLE
Get make windows working in CI

### PR DESCRIPTION
**Pull request by michael-loki**

### CODE REVIEW

File: cmake/win32.cmake [hunk 1]
```diff
@@ -6,10 +6,11 @@ enable_language(RC)
 set(CMAKE_CXX_STANDARD_LIBRARIES "${CMAKE_CXX_STANDARD_LIBRARIES} -lshlwapi")
 
 set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
-add_compile_options(/EHca /arch:AVX2 /MD)
-add_definitions(-D_SILENCE_CXX17_OLD_ALLOCATOR_MEMBERS_DEPRECATION_WARNING)
 
-if(NOT MSVC_VERSION)
```
Comment (despair86) [inline cmake/win32.cmake]: i have this fixed on my end

File: llarp/util/fs.hpp [hunk 2]
```diff
@@ -8,10 +8,7 @@
 #define PATH_SEP "/"
 #endif
 
-#ifdef WIN32
```
Comment (despair86) [inline llarp/util/fs.hpp]: ditto

File: llarp/util/threading.hpp [hunk 3]
```diff
@@ -5,7 +5,7 @@
 #include <absl/synchronization/mutex.h>
 #include <absl/time/time.h>
 
-#ifdef WIN32
+#if defined(WIN32) && !defined(__GNUC__)
```
Comment (despair86) [inline llarp/util/threading.hpp]: yup

File: vendor/abseil-cpp/absl/synchronization/internal/waiter.h [hunk 4]
```diff
@@ -19,7 +19,7 @@
 #include "absl/base/config.h"
 
 #ifdef _WIN32
-#include <Windows.h>
+#include <windows.h>
```
Comment (despair86) [inline vendor/abseil-cpp/absl/synchronization/internal/waiter.h]: this too

**Comment by despair86** [comment]
works on my machine, can't seem to reproduce: https://asciinema.org/a/iKl25hX0fxamk3fg0dDM5MyLf

**Comment by despair86** [comment]
@michael-loki try pulling from my repo, see if that works 🤷‍♂
if so i'll get my PR in

**Comment by michael-loki** [comment]
@majestrate ready for review

File: llarp/util/fs.hpp [hunk 5]
```diff
@@ -8,15 +8,12 @@
...
-#include <filesystem>
-namespace fs = std::filesystem;
-#elif defined(LOKINET_USE_CPPBACKPORT)
+#if defined(LOKINET_USE_CPPBACKPORT)
 #include <filesystem.h>
 namespace fs = cpp17::filesystem;
 #else
-#include <experimental/filesystem>
-namespace fs = std::experimental::filesystem;
+#include <filesystem>
```
Comment (majestrate) [inline llarp/util/fs.hpp]: linux builds use `<experimental/filesystem>`

**Comment by michael-loki** [comment]
@majestrate ready to merge

**Comment by michael-loki** [comment]
@despair86 these changes pass CI, lets get them in?

**Pull request merged**